src/python/pants/backend/javascript/README.md

```diff
@@ -0,0 +1,89 @@
+# Javascript Backend
+
+This README is mainly for the purposes of understanding the current backend functionality for PR review purposes, 
+plan to migrate most of this to user docs once that is ready. 
+
+## Definitions
+
+- Module: A single javascript source file. See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Modules
+- Package: A javascript directory (or file) described by a `package.json` file. See https://docs.npmjs.com/about-packages-and-modules
+- Workspaces: Functionality of package managers for supporting development of multiple packages in a single repository
+  - In [npm](https://docs.npmjs.com/cli/v7/using-npm/workspaces) and [yarn](https://classic.yarnpkg.com/lang/en/docs/workspaces/), a "workspaces" key is added to a top level `package.json` file, which points to child packages. 
+    - A single 'workspace' and 'package' have the same root directory, however, 'workspace' has an additional meaning in that package managers allow you to execute commands in the context of a workspace.
+    - The term 'workspace' (as singular) is not frequently used though.
+  - In [pnpm](https://pnpm.io/workspaces), a 'workspace' is defined by a `pnpm-workspace.yaml` file which points to child packages.
+    - i.e., a single workspace is linked to multiple child packages, unlike in the definition from npm and yarn. This is only trivial for our purposes, but worth keeping in mind when talking about workspaces.
+- Project: Not formally defined. It is often used informally to refer to packages (e.g. where a single package might be a React project). 
+  - In this backend, `ProjectEnvironment` is used to indicate the context (package + configuration) in which a node process is run. This could be inside a singular package directory, or a root package.json defining workspaces, or a directory containing a `pnpm-workspace.yaml` file. 
+
+## Repo Structure
+
+- repositoryRoot
+  - area-A/
+    - package.json with 'workspaces' defined (yarn, npm)
+      - workspaces: ["packages/package-a", "packages/package-b", "packages/sub-dir/package-c"]
+    - \[pnpm-workspace.yaml\] - pnpm only - will cause issues if present for npm/yarn
```

Comment:
Same here, link the issue. This shouldn't be a hard problem to fix. https://github.com/pantsbuild/pants/issues/21134

Reply:
I removed the README as I figured it would add a lot of noise to this PR at this stage, and it was mainly for me to work through things and clarify assumptions.  Prefer to use it as input to documentation instead, but can add here or separately if you think it's helpful for others.